src/parser/handle_heredoc.c

```diff
@@ -107,13 +93,13 @@ int	handle_heredoc(const char *delimiter)
 			print_heredoc_warning(delimiter);
 			break ;
 		}
-		if (ft_strcmp(line, delimiter) == 0)
-		{
-			free(line);
+		if (ft_strcmp(line, expanded) == 0)
 			break ;
-		}
 		heredoc_size += ft_strlen(line) + 1;
 		ft_lstadd_back(&input_list, ft_xlstnew(line));
 	}
-	return (open_heredoc(input_list, heredoc_size));
+	set_redirect_heredoc_info(info, input_list, heredoc_size,
+		delimiter[0] == expanded[0]);
```

Comment:
```bash
$cat << E"OF"
> $HOME
> EOF
$HOME
$
```
to fullfill this test case, it has to be like this:
```suggestion
		!ft_strcmp(delimiter, expanded));
```